ui/handler.go

```diff
@@ -121,6 +121,94 @@ func (h *internalHandler) handleNewRulesetRequest(w http.ResponseWriter, r *http
 
 }
 
+func (h *internalHandler) handleEditRulesetRequest(w http.ResponseWriter, r *http.Request) {
+	path := strings.TrimPrefix(r.URL.Path, "/rulesets/")
+	if path == "" {
+		writeError(w, r, nil, http.StatusNotFound)
+		return
+	}
+
+	nrr := &newRulesetRequest{}
+
+	err := json.NewDecoder(r.Body).Decode(nrr)
+	if err != nil {
+		writeError(w, r, err, http.StatusInternalServerError)
+		return
+	}
+
+	// Get the existing entry
+	entry, err := h.service.Get(r.Context(), path, "")
+	if err != nil {
+		if err == store.ErrNotFound {
+			writeError(w, r, err, http.StatusNotFound)
+			return
+		}
+
+		writeError(w, r, err, http.StatusInternalServerError)
+		return
+	}
+
+	// Update the entry with the new rules
+	err, status := updateEntry(entry, nrr)
+	if err != nil {
+		writeError(w, r, err, status)
+		return
+	}
+
+	// Write the new entry back to the DB
+	result, err := h.service.Put(r.Context(), path, entry.Ruleset)
+	if err != nil {
+		if err == store.ErrNotFound {
+			writeError(w, r, err, http.StatusNotFound)
+			return
+		}
+
+		writeError(w, r, err, http.StatusInternalServerError)
+		return
+	}
+	if result == nil {
+		err := fmt.Errorf("No Ruleset found at path: %q", path)
+		writeError(w, r, err, http.StatusNotFound)
+		return
+	}
+	reghttp.EncodeJSON(w, r, nil, http.StatusNoContent)
```

Comment:
Do we need to call `EncodeJSON` for that?

Does that work too?

```go
w.WriteHeader(http.StatusNoContent)
```

Reply:
:+1: